### PR TITLE
[TASK] Overhaul other reserved toplevel objects page

### PR DESCRIPTION
**Pull request by linawolf**
* Use confvals
* Correct indentation
* Improve descriptions
* Add examples

releases: main, 12.4, 11.5

### CODE REVIEW

File: Documentation/TopLevelObjects/Other.rst [hunk 1]
```diff
@@ -1,91 +1,135 @@
...
-         cached! Therefore use these names to store temporary data.
+..  confval:: temp
+    :name: tlo-temp
 
-         :typoscript:`lib` can be used for a "library" of code, you can reference in
-         TypoScript (unlike :typoscript:`styles` which is unset).
+    This top-level object name is reserved.
 
+    The top level object :typoscript:`temp` is used to store and copy
+    TypoScript code during parse-time.
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:23]:
```suggestion
    TypoScript code during parse time.
```

File: Documentation/TopLevelObjects/Other.rst [hunk 2]
```diff
@@ -1,91 +1,135 @@
...
+..  index:: Top-level objects; lib
 
-.. container:: table-row
+..  _top-level-objects-lib:
+..  _tlo-lib:
 
-   Property
-         resources
+lib
+====
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:54]:
```suggestion
===
```

File: Documentation/TopLevelObjects/Other.rst [hunk 3]
```diff
@@ -1,91 +1,135 @@
...
-   Data type
-         readonly
+..  confval:: lib
+    :name: tlo-lib
 
-   Description
-         Resources in list (internal)
+    This top-level object name is reserved.
 
+    The top level object :typoscript:`lib` is used to store, copy and reference
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:61]:
```suggestion
    The top-level object :typoscript:`lib` is used to store, copy and reference
```

File: Documentation/TopLevelObjects/Other.rst [hunk 4]
```diff
@@ -1,91 +1,135 @@
...
+    :name: tlo-lib
 
-   Description
-         Resources in list (internal)
+    This top-level object name is reserved.
 
+    The top-level object :typoscript:`lib` is used to store, copy and reference
+    TypoScript code.
 
+    Other then :confval:`tlo-temp` is not unset before the template is cached,
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:64]:
```suggestion
    Other than :confval:`tlo-temp` is not unset before the template is cached,
```

File: Documentation/TopLevelObjects/Other.rst [hunk 5]
```diff
@@ -1,91 +1,135 @@
...
+    :name: tlo-lib
 
-   Description
-         Resources in list (internal)
+    This top-level object name is reserved.
 
+    The top-level object :typoscript:`lib` is used to store, copy and reference
+    TypoScript code.
 
+    Other than :confval:`tlo-temp` is not unset before the template is cached,
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:64]: This sentence is hard to understand (at least for me), I read it multiple times.

File: Documentation/TopLevelObjects/Other.rst [hunk 6]
```diff
@@ -1,91 +1,135 @@
...
+    It is preserved for historic reasons, use :confval:`tlo-temp` instead.
+
+..  index:: Top-level objects; tt_content
+..  _tlo-tt:
+..  _tlo-tt_content:
+
+tt_content
+==========
+
+The top level keyword :typoscript:`tt_content` is used to render content
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:107]:
```suggestion
The top-level keyword :typoscript:`tt_content` is used to render content
```

File: Documentation/TopLevelObjects/Other.rst [hunk 7]
```diff
@@ -1,91 +1,135 @@
...
-.. _top-level-objects-lib:
-.. _tlo-lib:
+..  include:: /Includes.rst.txt
+..  _top-level-objects-other-reserved-tlo-s:
 
-==================
-temp, styles & lib
-==================
+==========================
+Reserved top level objects
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:5]:
```suggestion
Reserved top-level objects
```

File: Documentation/TopLevelObjects/Other.rst [hunk 8]
```diff
@@ -1,91 +1,135 @@
...
-         cache. :typoscript:`temp` and :typoscript:`styles` are unset before the template is
-         cached! Therefore use these names to store temporary data.
+..  confval:: temp
+    :name: tlo-temp
 
-         :typoscript:`lib` can be used for a "library" of code, you can reference in
-         TypoScript (unlike :typoscript:`styles` which is unset).
+    This top-level object name is reserved.
 
+    The top level object :typoscript:`temp` is used to store and copy
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:22]:
```suggestion
    The top-level object :typoscript:`temp` is used to store and copy
```

File: Documentation/TopLevelObjects/Other.rst [hunk 9]
```diff
@@ -1,91 +1,135 @@
...
+    TypoScript code during parse time.
 
-.. index:: Top-level objects; tt_content
-.. _tlo-tt:
-.. _tlo-tt_content:
+    :typoscript:`temp` is unset before the template is cached, objects in it
+    can therefore not be referenced. Use :confval:`tlo-lib` for that purpose.
 
+Example: Use the top level object `temp` to copy code
```
Comment (brotkrueml) [inline Documentation/TopLevelObjects/Other.rst:28]:
```suggestion
Example: Use the top-level object `temp` to copy code
```

File: Documentation/TopLevelObjects/Other.rst (hunk 5 above)
Comment (linawolf) [reply Documentation/TopLevelObjects/Other.rst:64]: How would you formulate it?

**Pull request merged**